src/eva/vision/data/datasets/segmentation/consep.py

```diff
@@ -0,0 +1,155 @@
+            split: Dataset split to use. If `None`, the entire dataset is used.
+            width: Width of the patches to be extracted, in pixels.
+            height: Height of the patches to be extracted, in pixels.
+            target_mpp: Target microns per pixel (mpp) for the patches.
+            backend: The backend to use for reading the whole-slide images.
+            transforms: Transforms to apply to the extracted image & mask patches.
+            seed: Random seed for reproducibility.
+        """
+        self._split = split
+        self._root = root
+        self._seed = seed
+
+        self.datasets: List[wsi.WsiDataset]  # type: ignore
+
+        wsi.MultiWsiDataset.__init__(
+            self,
+            root=root,
+            file_paths=self._load_file_paths(split),
+            width=width,
+            height=height,
+            sampler=sampler,
+            target_mpp=target_mpp,
+            overwrite_mpp=0.25,
+            backend="pil",
+            image_transforms=transforms,
+        )
+
+    @property
+    @override
+    def classes(self) -> List[str]:
+        # TODO: summarize classes
+        return [
+            "other",
+            "inflammatory",
+            "healthyepithelial",
+            "dysplastic/malignant epithelial",
+            "fibroblast",
+            "muscle",
+            "endothelial",
+        ]
+
+    @property
+    @override
+    def class_to_idx(self) -> Dict[str, int]:
+        return {label: index for index, label in enumerate(self.classes)}
+
+    @override
+    def prepare_data(self) -> None:
+        _validators.check_dataset_exists(self._root, True)
+
+        if not os.path.isdir(os.path.join(self._root, "Train")):
+            raise FileNotFoundError(f"Train directory not found in {self._root}.")
+        if not os.path.isdir(os.path.join(self._root, "Test")):
+            raise FileNotFoundError(f"Test directory not found in {self._root}.")
+
+    @override
+    def validate(self) -> None:
+        _validators.check_dataset_integrity(
+            self,
+            length=self._expected_dataset_lengths.get(self._split),
+            n_classes=len(self.classes),
+            first_and_last_labels=((self.classes[0], self.classes[-1])),
+        )
+
+    @override
+    def filename(self, index: int) -> str:
+        return os.path.basename(self._file_paths[self._get_dataset_idx(index)])
+
+    @override
+    def __getitem__(self, index: int) -> Tuple[tv_tensors.Image, tv_tensors.Mask]:
+        return base.ImageSegmentation.__getitem__(self, index)
+
+    @override
+    def load_image(self, index: int) -> tv_tensors.Image:
+        image_array = wsi.MultiWsiDataset.__getitem__(self, index)
+        return functional.to_image(image_array)
+
+    @override
+    def load_mask(self, index: int) -> tv_tensors.Mask:
+        path = self._get_mask_path(index)
+        mask = scipy.io.loadmat(path)["type_map"]
```

Comment:
should we have create `io/mat.py`  (e.g. [here](https://github.com/kaiko-ai/eva/pull/544/files#diff-a32048a5d4095072c8ab037e715eed80aa237020e888612c44f7affc63c44f76)) for this for consistency with other input formats?

Reply:
good point. done